src/app/Components/shopping-list/shopping-list.component.html

```diff
@@ -0,0 +1,90 @@
+<div class="container py-5">
+  <div class="row">
+    <div class="col-md-12 p-0">
+      <form [formGroup]="shoppingForm">
+        <div class="row">
+          <div class="form-group col-md-6">
+            <label class="font-weight-bold">Name</label>
+            <input type="text" class="form-control" formControlName="name" />
+            <div
+              *ngIf="
+                shoppingForm.controls['name']?.invalid &&
+                shoppingForm.controls['name']?.touched
+              "
+              class="alert alert-danger mt-2"
+              role="alert"
+            >
+              Name is required!
```

Comment:
shouldn't have warning msgs if the user only touches but does not fill in the form, cuz this form is optional

Reply:
I added a cancel button to cancel the ad, and cancel the update if they want because I see the validation and giving the warning message is still necessary.